grafana-plugin/src/containers/GSelect/GSelect.tsx

```diff
@@ -91,13 +89,18 @@ const GSelect = observer((props: GSelectProps) => {
       const searchResult = model.getSearchResult(query);
       const items = Array.isArray(searchResult.results) ? searchResult.results : searchResult;
 
-      const options = items.map((item: any) => ({
-        value: item[valueField],
-        label: get(item, displayField),
-        imgUrl: item.avatar_url,
-        description: getDescription && getDescription(item),
-      }));
-      return options;
+      return items.reduce((options: any[], item: any) => {
```

Comment:
My thinking was that instead of `items.filter` and then `items.map` to use `items.reduce` as it's one single call instead of two.

Reply:
we should not be scare of using two iterations here because it's more more readable, imho `reduce` is for complicated cases only, when other approaches don't work. Anyway we decided to use `filterOptions` here...